src/pandablocks_ioc/_pvi.py

```diff
@@ -227,10 +233,18 @@ def create_pvi_records(record_prefix: str):
         # TODO: label widths need some tweaking - some are pretty long right now
         formatter = DLSFormatter(label_width=250)
 
-        screens_dir_contents = list(Pvi._screens_dir.iterdir())
-        if screens_dir_contents:
-            raise FileExistsError("Screens directory is not empty")
-
-        for device in devices:
-            bobfile_path = Pvi._screens_dir / Path(f"{device.label}.bob")
-            formatter.format(device, record_prefix + ":", bobfile_path)
+        if Pvi._screens_dir:
+            screens_dir_contents = list(Pvi._screens_dir.iterdir())
+            if screens_dir_contents:
+                if Pvi._skip_bobfile_gen_if_dir_not_empty:
```

Comment:
It would be useful to print a warning message telling that generation of bob files was skipped because the directory (with name) is not empty and the user should clear the directory and restart the IOC to generated the updated bob files.